js/isir-module-2025-2026.js

```diff
@@ -743,120 +351,166 @@ export const valid_country_codes = {
+    },
+    range: (sz_value, op) => _check_range(sz_value, op.min, op.max),
+    year: (sz_value) => _check_range(sz_value, 1900, 2100),
+    date: (sz_value) => _check_date(sz_value),
+    ssn: (sz_value, op) => (
+        /^\d{9}$/.test(sz_value) // 9 digits matching full string
+        && (sz_value >= op.min) // lexographically >= min value
+        && (sz_value <= op.max) // lexographically <= max value
+        // && {valid: true, result: 'ssn:'+sz_value}
+    ),
+    itin: (sz_value, op) => (
+        /^\d{9}$/.test(sz_value) // 9 digits matching full string
+        && (sz_value >= op.min) // lexographically >= min value
+        && (sz_value <= op.max) // lexographically <= max value
+        // && {valid: true, result: 'itin:'+sz_value}
+    ),
+    school_code: (sz_value) => /^[0BEG]\d{5}$|^$/.test(sz_value),
+    region_code: (sz_value) => /^\d+$/.test(sz_value),
+    dhs_case_number: (sz_value) => /[0-9]{13}[A-Z]{2}/.test(sz_value),
+    eti_destination: (sz_value) => /(:?FT|TG)[0-9A-Za-z]{5}/.test(sz_value),
+    comment_codes: (sz_value) => {
+        let result = sz_value.split(/(\d\d\d)/) // split by triple-digits
+            .filter((v,i) => i&1); // keep only the split options
 
-  // Canada and Mexico state codes
-  'CN': 1, // Canada
-  'MX': 1, // Mexico
+        // valid all codes combined equals the entire value
+        let valid = result.join('') == sz_value;
+        return valid && {valid, result}
+    },
+    reject_codes: (sz_value) => {
+        let result = sz_value.split(/(\d\d|\d\s|\d$)/) // split by two-digits
+            .filter((v,i) => i&1); // keep only the split options
 
-  // FC for Foreign Country as State Code per table 4-4 
-  // of Volume 4, Record Layouts and Processing Codes
-  'FC': 1, // Foreign Country
+        // valid all codes combined equals the entire value
+        let valid = result.join('') == sz_value;
+        return valid && {valid, result}
+    },
+};
+
+function _option_for(sz_value, field) {
+    let op, res;
+    for (op of field.options) {
+        if (null == _validate_by_op[op.op]) {
+            console.warn('Missing validate op', op);
+            continue
+        }
+        res = _validate_by_op[op.op](sz_value, op, field);
+        if (res) {
+            if (Object.hasOwn(res, 'result'))
+                res.value = sz_value;
+            return res
+        }
+    }
+}
+function _validate_options(sz_value, field) {
+    let res;
+    if (field.opt_len) {
+        // transform into list of `opt_len` character groups
+        let opt_list = sz_value
+            .split(new RegExp(`(.{${field.opt_len}})`))
+            .filter((v,i) => i&1) // keep only the split options
+            .map(opt_val => _option_for(opt_val, field));
+        if (opt_list.length > 0)
+            res = {valid: opt_list.every(v=>v.valid), value: sz_value, result: opt_list};
+    } else {
+        res = _option_for(sz_value, field);
+    }
+
+    if (null != res)
```

Comment:
Is this check meant to cover when res is defined? `null != undefined` returns false. There are two better options for this check. For native javascript, we can check `res !== undefined`, or if we can grabe a lodash package it does a good job of covering more cases, e.g. [`isUndefined`](https://lodash.com/docs#isUndefined) .